crates/bevy_core_pipeline/src/tonemapping/mod.rs

```diff
@@ -128,35 +250,107 @@ pub fn queue_view_tonemapping_pipelines(
     pipeline_cache: Res<PipelineCache>,
     mut pipelines: ResMut<SpecializedRenderPipelines<TonemappingPipeline>>,
     upscaling_pipeline: Res<TonemappingPipeline>,
-    view_targets: Query<(Entity, &Tonemapping)>,
+    view_targets: Query<(Entity, Option<&Tonemapping>, Option<&DebandDither>), With<ViewTarget>>,
 ) {
-    for (entity, tonemapping) in view_targets.iter() {
-        if let Tonemapping::Enabled { deband_dither } = tonemapping {
-            let key = TonemappingPipelineKey {
-                deband_dither: *deband_dither,
-            };
-            let pipeline = pipelines.specialize(&pipeline_cache, &upscaling_pipeline, key);
-
-            commands
-                .entity(entity)
-                .insert(ViewTonemappingPipeline(pipeline));
-        }
+    for (entity, tonemapping, dither) in view_targets.iter() {
+        let key = TonemappingPipelineKey {
+            deband_dither: *dither.unwrap_or(&DebandDither::Disabled),
+            tonemapping: *tonemapping.unwrap_or(&Tonemapping::None),
+        };
+        let pipeline = pipelines.specialize(&pipeline_cache, &upscaling_pipeline, key);
+
+        commands
+            .entity(entity)
+            .insert(ViewTonemappingPipeline(pipeline));
     }
 }
-
-#[derive(Component, Clone, Reflect, Default, ExtractComponent)]
+/// Enables a debanding shader that applies dithering to mitigate color banding in the final image.
+#[derive(
+    Component,
+    Debug,
+    Hash,
+    Clone,
+    Copy,
+    Reflect,
+    Default,
+    ExtractComponent,
+    PartialEq,
+    Eq,
+    FromReflect,
+)]
 #[extract_component_filter(With<Camera>)]
 #[reflect(Component)]
-pub enum Tonemapping {
+pub enum DebandDither {
     #[default]
     Disabled,
-    Enabled {
-        deband_dither: bool,
-    },
+    Enabled,
 }
 
-impl Tonemapping {
-    pub fn is_enabled(&self) -> bool {
-        matches!(self, Tonemapping::Enabled { .. })
-    }
+pub fn get_lut_bindings<'a>(
```

Comment:
Should these be pub? Definitely feels like a fragile implementation detail to me.

Reply:
These are helpful if someone wants to include tonemapping in their own custom material that doesn't use AsBindGroup or in a custom post process node. The environment map plugin provides similar pub binding/layout functions.

Reply:
Alright 👍🏽